### ISSUE TITLE
Subdivisions

### ISSUE THREAD

**Issue by puschie286** [nager/Nager.Country#30]
Hi,

what do you think about adding subdivision (ISO 3166-2) information ?

We need that for some project so im currently implementing it (europe + usa for now). Its basically an ISubdivisionInfo with informations from [wikipedia](https://en.wikipedia.org/wiki/ISO_3166-2:AT) and [iso obp](https://www.iso.org/obp/ui/#iso:code:3166:AT) and an subdivision list on country info.

If you are interested, i would prepare an PR with structure and some reference definitions.

here is our structure so far:
```
public interface ISubdivisionInfo
{
	/// <summary>
	/// CommonName
	/// </summary>
	string CommonName { get; }
	/// <summary>
	/// NativeName
	/// </summary>
	string OfficialName { get; }
	/// <summary>
	/// ISO-3166-2 code
	/// </summary>
	string TextCode { get; }
	/// <summary>
	/// Subdivision code
	/// </summary>
	SubdivisionCode Code { get; }
	/// <summary>
	/// Subdivison category
	/// </summary>
	SubdivisionCategory Category { get; }
}
```
CommonName is the reference name (from wikipedia, mostly english)
OfficialName is the iso name (mostly local)
SubdivisionCode is an enum with all subdivison codes (without the - ) from [here](https://www.ip2location.com/free/iso3166-2)

was thinking about adding administration language but it seems that you dont have this on the CountryInfo and the LanguageCode enum is part of the translation project

**Comment by tinohager** [comment]
Yes, I could imagine that. My suggestion would be to implement it like Nager.Country.Translation simply as an additional module to Nager.Country. This would not increase the size of the main package.

**Comment by tinohager** [comment]
@puschie286 is there any news here yet?

**Comment by puschie286** [comment]
@tinohager looks good so far, we have a working solution in use for a few weeks now. It may have some unusual (related to the your base project) things like fixed enum values or extension methods.
Created a PR for this state #33 with all relevant changes.
If you like to have the generators for the subdivision files and initializer classes, i can add them too or send them - didnt see them in the country project so i left them out

**Comment by puschie286** [comment]
We dropped the idea of subdivision as separate project, so im closing this issue

**Comment by tinohager** [comment]
All right, thanks for your feedback. Maybe I'll add it when I get the chance. But then I would like to implement it for all countries.
